xamarin/Xamarin.Forms
# [UWP] fix performance on UWP ListView if ItemsSource is not an ObservableCollection

### Description of Change ###

If the ItemSource ListView implements the IList collection and INotifyCollectionChanged, then this collection does not wrap into an ObservableCollection.

### Issues Resolved ### 

- fixes #4059 

### API Changes ###

 None

### Platforms Affected ### 

- UWP

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 

**Before**
![screenshot_2](https://user-images.githubusercontent.com/27482193/47150571-9be7cd80-d2df-11e8-9207-0bd249364436.png)

**After**
![screenshot_1](https://user-images.githubusercontent.com/27482193/47149837-6e018980-d2dd-11e8-9f2c-65cbfab2918e.png)

### Testing Procedure ###

- run UI test 3273
- check that less than 1000 items were loaded in the right list
- items can be drag&drop in the right list
- when scrolling the list, more items should be dynamically loaded

### PR Checklist ###
